packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -66,15 +66,36 @@ const ProductTile = (props) => {
         ...rest
     } = props
 
-    const {currency, image, price, productId, hitType} = product
+    const {image, productId, hitType} = product
+    // NOTE: swatches will implement later to set variant accordingly,
+    // for now we use the first variant that has price book to set up discount price
+    // Not all variants has set in a priceBook, a.k.a not having tieredPrices.
+    const variant = product?.variants?.find((i) => !!i?.tieredPrices)
+    let currentPrice = variant ? variant?.price : product?.price
 
     // ProductTile is used by two components, RecommendedProducts and ProductList.
     // RecommendedProducts provides a localized product name as `name` and non-localized product
     // name as `productName`. ProductList provides a localized name as `productName` and does not
     // use the `name` property.
     const localizedProductName = product.name ?? product.productName
+    const isProductASet = hitType === 'set' || !!product?.type?.set
```

Comment:
Do we need a check for product bundles?

Reply:
The code is already covered the bundle case implicitly, but I will refactor code to make it more transparent here. Thanks for the feedback.